x/lockup/keeper/lock.go

```diff
@@ -384,8 +393,13 @@ func (k Keeper) ExtendLockup(ctx sdk.Context, lock types.PeriodLock, newDuration
 		return fmt.Errorf("cannot edit lockup with synthetic lock %d", lock.ID)
 	}
 
-	oldLock := lock
+	// completely delete existing lock refs
+	err = k.deleteLockRefs(ctx, unlockingPrefix(lock.IsUnlocking()), *lock)
+	if err != nil {
+		return err
+	}
 
+	oldDuration := lock.GetDuration()
 	if newDuration != 0 {
 		if newDuration <= lock.Duration {
```

Comment:
```suggestion
		if newDuration <= oldDuration {
```
If we're defining it right above =p